### DISCUSSION TITLE
Simple Cyberpunk 2077 theme + FPS meter

### DISCUSSION THREAD

**Discussion by pingyaru** [mathoudebine/turing-smart-screen-python#367] [Themes]
This theme is based on [Jojo10011's Simple Dark Theme](https://github.com/mathoudebine/turing-smart-screen-python/discussions/195) 

I just tweaked a little things + added FPS meter. I found it kind of difficult to read small text on 3.5" screen, so there's no additional information.

![preview](https://github.com/mathoudebine/turing-smart-screen-python/assets/147753559/d73c2031-82f9-4048-a5b0-77dd2595d52a)

![photo_2023-10-22_16-16-53](https://github.com/mathoudebine/turing-smart-screen-python/assets/147753559/fcf2471b-c727-44ab-b7d4-dafbfd7c5e79)

Also, you can find .psd file and mesh grid in the folder.

![mesh](https://github.com/mathoudebine/turing-smart-screen-python/assets/147753559/308272cd-4765-4a93-b584-5c561a7b8bc4)

**Download:** [Landscape_simple_cyberpunk.zip](https://github.com/mathoudebine/turing-smart-screen-python/files/13063339/Landscape_simple_cyberpunk.zip)

**Comment by Agudelo91** [comment]
Hello,
Nice one! but it says "Your GPU FPS is not supported yet".
I have a 4070Ti, is there a way to fix it?
Also it takes the CPU temperature from a Core, not the Package one, do you know the reason of it?
Thanks!

**Reply by pingyaru** [reply]
Hello!
Thank you =)

Unfortunately, there's no support for Nvidia GPUs. Probably there'll be such option in future versions. Currently it works only for AMD GPUs.

What about CPU temperature, probably it's the way the author coded the program. Unfortunately, I don't know whether it's possible to change the temperature source without changing the code.

**Reply by mathoudebine** [reply]
Hello! Unfortunately the FPS meter is only available for AMD GPUs for now. Check this document for more details: https://github.com/mathoudebine/turing-smart-screen-python/wiki/System-monitor-%3A-hardware-support
This is a limitation from the [LibreHardwareMonitor](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor) library we use on Windows to get hardware sensors.

For the CPU temperature, this code is used:
https://github.com/mathoudebine/turing-smart-screen-python/blob/d44505f03ab7f35cbb9f7eb9bf3bed86d182d458/library/sensors/sensors_librehardwaremonitor.py#L217-L237
We try to use "Core Average" temperature if available, if not then we default to "Core Max" or "CPU Package".
If you want you can reorder the lines to use "CPU Package" sensor in priority.

**Reply by Agudelo91** [reply]
Thank you both for your replays :)

**Reply by xfridgewizardx** [reply]
Mine is displaying CPU Package but I want Core.  Why is it bypassing the first 2 and going straight for CPU Package?

**Reply by mathoudebine** [reply]
Can you run the `external/LibreHardwareMonitor/test_librehardwaremonitor.py` script and post the result here?

**Reply by xfridgewizardx** [reply]
Please see attached!
[test_librehardwaremonitor_results.txt](https://github.com/mathoudebine/turing-smart-screen-python/files/13343402/test_librehardwaremonitor_results.txt)

**Reply by xfridgewizardx** [reply]
Also these are my temps from the Asus Armoury Crate Software.

![Screenshot 2023-11-13 190624](https://github.com/mathoudebine/turing-smart-screen-python/assets/72410588/cf04e5e2-f9a0-4aa5-84c7-3fc5b8057e8b)

**Reply by cesarhfborges** [reply]
> Also these are my temps from the Asus Armoury Crate Software.
> 
> ![Screenshot 2023-11-13 190624](https://private-user-images.githubusercontent.com/72410588/282632429-cf04e5e2-f9a0-4aa5-84c7-3fc5b8057e8b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTcxNjEsIm5iZiI6MTczNDM1Njg2MSwicGF0aCI6Ii83MjQxMDU4OC8yODI2MzI0MjktY2YwNGU1ZTItZjlhMC00YWE1LTg0YzctM2ZjNWI4MDU3ZThiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE2VDEzNDc0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxZDAyMjQ2MDQxOTE2MjczMjg1OTkzOTU1YjVlZjBjOTNhMGNjMTlmZGI1ZmVhMjA0M2ZhYzY3MzNiYjExOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ghq62uWKgq-VLG1V_k1v7peKOtQzSwvIHiujclZR6Rg)

Like this:
https://github.com/mathoudebine/turing-smart-screen-python/discussions/608

**Reply by aStroNuts04** [reply]
Hi ive heard afterburner with rivatuner can be used to display fps for the turzx screens. Anyone figured out how to do it?

**Comment by mathoudebine** [comment]
Thanks for the theme and also for sharing the resources!

**Comment by DylanGegax** [comment]
Really love this theme just I seem to be having a problem with the GPU side of things sadly. The FPS meter doesn't seem to be working and the GPU % and temp update like once every 4 mins. Any help would be appreciated :)

**Reply by mathoudebine** [reply]
Just in case other people want to use this theme: there is an issue with the polling interval for the GPU data in this theme.
The `GPU` section from the `theme.yaml` file must contain an `INTERVAL` in seconds:
Edit the theme from:
![image](https://github.com/user-attachments/assets/02b1101f-3e24-4c89-9518-7b51828d51e7)
to:
![image](https://github.com/user-attachments/assets/4c0eef49-cc19-4773-9ea0-76880f10a783)
Be careful of the identation!

**Reply by DylanGegax** [reply]
Hi! I'm not sure if I did it right but I did try doing that before and it didn't work sadly but it didn't break the whole thing so I just kept it like that. Would there be anyway for you to drop a link to a modified version thats correct? I really appreciate the help with this!! Heres what mine looks like at the moment:

![gpu interval](https://github.com/user-attachments/assets/98bc03c5-e2eb-4c6f-8586-beb9f7872930)

**Comment by Schaoki** [comment]
Hey,

i have a stupid question i guess.

how can i load it in the usbmonitor from turzx. i only can load there .data files

Would be nice if anyone can give me help there :)

Thank you

**Reply by c0ldJS** [reply]
You'll have to manually recreate it using TURZX's visual editing tool.

The themes for this project's system monitor cannot be lifted-and-shifted into the native TURZX monitor.

**Comment by Deny1006** [comment]
Any chance you can post version with date and time instead of FPS? I was trying for over than hour but probably it isn't as simple as copying and pasting from different theme.

**Reply by c0ldJS** [reply]
Try the following changes in the theme's YAML file (line numbers added for reference/convenience) and salt to taste:

```
53  CHRONO:
54    TEXT: DATE | TIME
55    X: 360
56    Y: 190
57    FONT: generale-mono/GeneraleMonoA.ttf
58    FONT_SIZE: 20
59    FONT_COLOR: 255, 239, 8
60    BACKGROUND_IMAGE: background.png
```

```
141  DATE:
142    INTERVAL: 1
143    DAY:
144      TEXT:
145        FORMAT: long
146        SHOW: True
147        X: 325
148        Y: 218
149        FONT: generale-mono/GeneraleMonoA.ttf
150        FONT_SIZE: 30
151        FONT_COLOR: 2, 216, 243
152        BACKGROUND_IMAGE: background.png
153    HOUR:
154      TEXT:
155        FORMAT: short
156        SHOW: True
157        X: 327
158        Y: 250
159        FONT: generale-mono/GeneraleMonoA.ttf
160        FONT_SIZE: 28
161        FONT_COLOR: 2, 216, 243
162        BACKGROUND_IMAGE: background.png
```

**Reply by Deny1006** [reply]
Thank you for that tip, I will try today

**Comment by GroGaloraGran** [comment]
An interesting idea, but for some reason the FPS does not show (
